packages/hardhat/contracts/Staking.sol

```diff
@@ -115,18 +147,34 @@ contract Staking is Ownable, ReentrancyGuard {
         if (userShares[user] == 0) return 0;
 
         // Calculate current exchange rate with accrued interest
-        uint256 currentExchangeRate = exchangeRate;
-        if (totalShares > 0) {
-            uint256 timeElapsed = block.timestamp - lastUpdateTime;
-            if (timeElapsed > 0) {
-                uint256 totalValue = (totalShares * exchangeRate) / PRECISION;
-                uint256 interest = (totalValue * interestRate * timeElapsed) / (SECONDS_PER_YEAR * 10000);
-                if (interest > 0) {
-                    currentExchangeRate += (interest * PRECISION) / totalShares;
-                }
-            }
-        }
+        uint256 currentExchangeRate = _getCurrentExchangeRate();
 
         return (userShares[user] * currentExchangeRate) / PRECISION;
     }
+
+    function _getCurrentExchangeRate() internal view returns (uint256) {
```

Comment:
I like how you put this logic in a helper function. It would be nice if we could add a method so that a user can see their current staked MyUSD balance (not in shares but in actual value).